org.testeditor.web.dropwizard/src/main/java/org/testeditor/web/dropwizard/auth/DefaultAuthDynamicFeature.xtend

```diff
@@ -1,45 +1,44 @@
 package org.testeditor.web.dropwizard.auth
 
-import javax.ws.rs.container.ContainerRequestFilter
+import javax.inject.Inject
 import javax.ws.rs.container.DynamicFeature
 import javax.ws.rs.container.ResourceInfo
 import javax.ws.rs.core.FeatureContext
+import org.testeditor.web.dropwizard.DropwizardApplicationConfiguration
 
 /**
- * make sure that auth filter is applied to all methods that do not have
- * '@NoAuth' annotations. annotating the class will disable authoristion
- * for all methods
- * 
+ * All methods are protected by default and are accessible with valid jwt token only!
+ * Methods explicitly annotated with @ApiTokenAuth are accessible with api token only!
+ * Methods explicitly annotated with @NoAuth are accessible without authentication
+ *  
  * see http://www.dropwizard.io/1.2.0/docs/manual/auth.html#chained-factories
  * and http://dinomite.net/blog/2016/05/18/optional-authentication-with-dropwizard/
  */
 class DefaultAuthDynamicFeature implements DynamicFeature {
 
-	val ContainerRequestFilter authFilter
-	val Class<? extends ContainerRequestFilter> authFilterClass
+	DropwizardApplicationConfiguration configuration
 
-	new(ContainerRequestFilter authFilter) {
-		this.authFilter = authFilter
-		this.authFilterClass = null
-	}
-
-	new(Class<? extends ContainerRequestFilter> authFilterClass) {
-		this.authFilter = null
-		this.authFilterClass = authFilterClass
-	}
+	@Inject JWTAuthFilter.Builder authFilterBuilder
+	@Inject ApiTokenAuthFilter.Builder apiTokenAuthFilterBuilder
 
 	override configure(ResourceInfo resourceInfo, FeatureContext context) {
-		if (resourceInfo.resourceMethod.getAnnotation(NoAuth) === null && resourceInfo.resourceClass.getAnnotation(NoAuth) === null) {
-			registerAuthFilter(context)
+		val method = resourceInfo.resourceMethod
+		val clazz = resourceInfo.resourceClass
+		val annotatedWithNoAuth = method.getAnnotation(NoAuth) !== null || clazz.getAnnotation(NoAuth) !== null
```

Comment:
Both method or clazz might be null according to the spec. We could also extract this easily to a method called something like `isResourceAnnotatedWith(ResourceInfo, Class<? extends Annotation>)`